lib/chef/knife/uploader_data_bag.rb

```diff
@@ -89,10 +91,16 @@ def diff_data_bag_item(item, item_id, old_attributes, new_attributes, diff_comme
         return false
       end
 
-      ui.info("#{diff_comment_prefix} data bag item #{item_id} " +
-              "(#{old_could_decrypt ? 'decrypted' : 'raw'} #{desc1} vs." +
-              " #{new_could_decrypt ? 'decrypted' : 'raw'} #{desc2}):\n" +
-              diff_color(old_decrypted_formatted, new_decrypted_formatted) + "\n")
+      if !show_encrypted && (old_could_decrypt || new_could_decrypt)
+        ui.info("#{diff_comment_prefix} data bag item #{item_id}. \n" +
+                "\tNot showing the diff since they are encrypted.\n" +
+                "\tUse --show_encrypted to display the diff.\n\n")
+      else
+        ui.info("#{diff_comment_prefix} data bag item #{item_id} " +
+                "(#{old_could_decrypt ? 'decrypted' : 'raw'} #{desc1} vs." +
+                " #{new_could_decrypt ? 'decrypted' : 'raw'} #{desc2}):\n" +
+                diff_color(old_decrypted_formatted, new_decrypted_formatted) + "\n")
```

Comment:
We should probably still say something about whether we could decrypt one or both databags. If we know both of them are encrypted, then trying to diff post-encryption data is not very useful.